receiver/otlpreceiver/metrics/otlp_test.go

```diff
@@ -0,0 +1,279 @@
+	require.NoError(t, err, "Failed to export metrics: %v", err)
+	require.NotNil(t, resp, "The response is missing")
+
+	// assert
+
+	require.Equal(t, 1, len(metricSink.metricsDataList),
+		"unexpected length: %v", len(metricSink.metricsDataList))
+
+	metricsData := consumerdata.MetricsData{
+		Node: &occommon.Node{},
+		Resource: &ocresource.Resource{
+			Labels: map[string]string{
+				"key1": "value1",
+			},
+		},
+		Metrics: []*ocmetrics.Metric{
+			{
+				MetricDescriptor: &ocmetrics.MetricDescriptor{
+					Name:        "mymetric",
+					Description: "My metric",
+					Unit:        "ms",
+					Type:        ocmetrics.MetricDescriptor_CUMULATIVE_INT64,
+					LabelKeys: []*ocmetrics.LabelKey{
+						{Key: "key1"},
+						{Key: "key2"},
+					},
+				},
+				Timeseries: []*ocmetrics.TimeSeries{
+					{
+						StartTimestamp: unixnanoToTimestamp(unixnanos1),
+						LabelValues: []*ocmetrics.LabelValue{
+							{
+								// key1
+								Value:    "value1",
+								HasValue: true,
+							},
+							{
+								// key2
+								HasValue: false,
+							},
+						},
+						Points: []*ocmetrics.Point{
+							{
+								Timestamp: unixnanoToTimestamp(unixnanos2),
+								Value: &ocmetrics.Point_Int64Value{
+									Int64Value: 123,
+								},
+							},
+						},
+					},
+					{
+						StartTimestamp: unixnanoToTimestamp(unixnanos1),
+						LabelValues: []*ocmetrics.LabelValue{
+							{
+								// key1
+								HasValue: false,
+							},
+							{
+								// key2
+								Value:    "value2",
+								HasValue: true,
+							},
+						},
+						Points: []*ocmetrics.Point{
+							{
+								Timestamp: unixnanoToTimestamp(unixnanos2),
+								Value: &ocmetrics.Point_Int64Value{
+									Int64Value: 456,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	assert.EqualValues(t, metricsData, metricSink.metricsDataList[0])
+}
+
+func makeMetricsServiceClient(port int) (collectormetrics.MetricsServiceClient, func(), error) {
+	addr := fmt.Sprintf(":%d", port)
+	cc, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	metricsClient := collectormetrics.NewMetricsServiceClient(cc)
+
+	doneFn := func() { _ = cc.Close() }
+	return metricsClient, doneFn, nil
+}
+
+func otlpReceiverOnGRPCServer(t *testing.T, mc consumer.MetricsConsumer) (r *Receiver, port int, done func()) {
+	ln, err := net.Listen("tcp", "localhost:")
+	require.NoError(t, err, "Failed to find an available address to run the gRPC server: %v", err)
+
+	doneFnList := []func(){func() { ln.Close() }}
+	done = func() {
+		for _, doneFn := range doneFnList {
+			doneFn()
+		}
+	}
+
+	_, port, err = hostPortFromAddr(ln.Addr())
+	if err != nil {
+		done()
+		t.Fatalf("Failed to parse host:port from listener address: %s error: %v", ln.Addr(), err)
+	}
+
+	r, err = New(receiverTagValue, mc)
+	require.NoError(t, err, "Failed to create the Receiver: %v", err)
```

Comment:
Nit: require will abort the test (I think it is identical to calling t.Fatal), however we do not call `done()` in this codepath while we do that above before calling t.Fatal. In practice what this probably means is that we will leak a tcp port during test, but probably not a big deal since it is a dynamically allocated port.
I do not think cleanup in the tests is important provided that we do not leak anything that is statically allocated and can affect other tests that follow.